src/backend/api/meals.js

```diff
@@ -4,8 +4,94 @@ const knex = require("../database");
 
 router.get("/", async (request, response) => {
   try {
-    // knex syntax for selecting things. Look up the documentation for knex for further info
-    const titles = await knex("meals").select("title");
+    if (request.query.limit && request.query.maxPrice){
+      const limit=request.query.limit;
+      const maxPrice=request.query.maxPrice;
+      const limitedMeals=await knex("meals").where("price","<=",maxPrice).limit(limit);
+
+    response.json(limitedMeals)
+    }
+    else if(request.query.availableReservations){
+      
+      const availableReservations=request.query.availableReservations;
+      const cheapMeal=await knex("meals").where("max_reservations",">",0)
```

Comment:
Does the meals table has a column "max_reservations"? I believe what the objective here is to join the meals table with the reservations table and find out the available meals based on "number_of_guests" a reservation has and the max_available_guests! If you need any tips, let me know so I can help :)

Reply:
hey, can you pls help me with this part?

Reply:
Absolutely! So, one approach is to 1) find all the meals ids that have available reservations, and then 2) use them to filter the meals table and only select these ids.
So, a snippet to help you for 1) is:
```
const availableMealsIds = knex('meal')
    .select('meal.id', 'meal.max_reservations')
    .leftJoin('reservation', { 'meal.id': 'reservation.meal_id' })
    .count({'reservations' : 'reservation.meal_id'})
    .groupBy('meal.id')
    .having(knex.raw('meal.max_reservations > reservations'));
```
Then, don't forget to do 2) https://knexjs.org/#Builder-whereIn